### DISCUSSION TITLE
How to effectively embed PDFs with images for a RAG LLM?

### DISCUSSION THREAD

**Discussion by lucasmirachi** [facebookresearch/faiss#4031] [Q&A]
I’m working on a project where I need to embed PDF documents that contain images (which may or may not be relevant to the response) to create a vector database for later retrieval in a Retrieval-Augmented Generation (RAG) LLM. Currently, I’m using Unstructured + Faiss, but I’m not achieving satisfactory results with the images in the PDFs.

Here are some details about my approach:

- I’m using the Unstructured library to parse the PDFs.

- FAISS is being used to create and manage the vector database.

- Text embeddings are working fine, but image embeddings are not yielding good results.

Questions:

1. What are the best practices for embedding PDFs that contain both text and images?

2. Are there any specific techniques or libraries that handle image embeddings within PDFs more effectively?

3. How can I improve the integration of image embeddings with text embeddings in my current setup?

Any advice or suggestions would be greatly appreciated!

**Comment by andysingal** [comment]
> I’m working on a project where I need to embed PDF documents that contain images (which may or may not be relevant to the response) to create a vector database for later retrieval in a Retrieval-Augmented Generation (RAG) LLM. Currently, I’m using Unstructured + Faiss, but I’m not achieving satisfactory results with the images in the PDFs.
> 
> Here are some details about my approach:
> 
> * I’m using the Unstructured library to parse the PDFs.
> * FAISS is being used to create and manage the vector database.
> * Text embeddings are working fine, but image embeddings are not yielding good results.
> 
> Questions:
> 
> 1. What are the best practices for embedding PDFs that contain both text and images?
> 2. Are there any specific techniques or libraries that handle image embeddings within PDFs more effectively?
> 3. How can I improve the integration of image embeddings with text embeddings in my current setup?
> 
> Any advice or suggestions would be greatly appreciated!

i have added examples within: https://github.com/andysingal/llm-course

**Comment by hypomariia** [comment]
https://lu.ma/b1yqug98 - recently found a great webinar

**Comment by RoaminS** [comment]
It's like this i work a lot on you can have all in local?

https://github.com/RoaminS/Memory-os-ai.git

https://github.com/user-attachments/assets/6932c9e6-eb58-40dc-8390-121dbee6cc82

**Reply by RoaminS** [reply]
> It's like this i work a lot on you can have all in local?
> 
> https://github.com/RoaminS/Memory-os-ai.git
> 
>  localpdf.mp4

I just don' make GUI / UI but if some one is good to build a good one, it will be an amazing open source project
